packages/apputils-extension/src/index.ts

```diff
@@ -220,13 +222,17 @@ const themes: JupyterLabPlugin<IThemeManager> = {
 /**
  * The default window name resolver provider.
  */
-const resolver: JupyterLabPlugin<IWindowResolver> = {
+const resolver: JupyterFrontEndPlugin<IWindowResolver> = {
   id: '@jupyterlab/apputils-extension:resolver',
   autoStart: true,
   provides: IWindowResolver,
-  requires: [IRouter],
-  activate: async (app: JupyterLab, router: IRouter) => {
-    const resolver = new WindowResolver();
+  requires: [IRouter, JupyterLab.IInfo],
```

Comment:
As above, my sense is that if we are requiring `JupyterLab.IInfo`, we might want to restrict the plugin to only working with `JupyterLab`.

Reply:
You could imagine an app that's basically a clone of JupyterLab and not just a subset ...

But actually in this case, I'd really like it if we could make a window resolver available to other applications, because this is a tedious problem to have to solve. It just required more thoughts than I could hold in my head for this PR.

This particular plugin *only* wants `info.defaultWorkspace` so it's trivial to make it JupyterLab-independent once we come up with a good subset of `IInfo`.

Reply:
Sounds good to me. I think my concerns would be eliminated once we find that subset.